app/src/citations.py

```diff
@@ -33,28 +36,67 @@ def create_citation(self, chunk: Chunk, text: str, text_headings: Sequence[str])
 
 def default_chunk_splitter(
     chunk: Chunk, factory: CitationFactory = citation_factory
+) -> list[Subsection]:
+    try:
+        return tree_based_chunk_splitter(chunk, factory)
+    except RuntimeError as e:
+        logger.warning(
+            "Falling back to basic_chunk_splitter for chunk: %s", chunk.id, exc_info=True
+        )
+        logger.warning(e)
+        return basic_chunk_splitter(chunk, factory)
```

Comment:
Since this is run during user interaction, fall back to the basic splitter and log a warning to diagnose the error.